nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/controller/StandardProcessorNode.java

```diff
@@ -1090,6 +1090,16 @@ public int getActiveThreadCount() {
                     break;
                 }
             }
+
+            // Ensure that execution node will not be misused
+            if (getExecutionNode() == ExecutionNode.PRIMARY) {
+                if (getInputRequirement() == Requirement.INPUT_REQUIRED
+                        || getInputRequirement() == Requirement.INPUT_ALLOWED && !getIncomingConnections().isEmpty()) {
```

Comment:
I think this can be simplified to just: `if (getExecutionNode() == ExecutionNode.PRIMARY && hasIncomingConnection()) {`